### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by tiagorahal**
### TDD project

- [x] Create a class called Solver.
- [x] Create a method called factorial that takes one argument, an integer N, and returns the factorial for that number. 
- [x] Create a method called reverse that takes one argument, a string word, and returns word reversed.
- [x] Create a method called fizzbuzz that takes one argument, an integer N, and returns a string.

### CODE REVIEW

File: spec/solver_test.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+require_relative '../solver'
+describe Solver do
+  before(:context) do
+    @solver = Solver.new
+  end
+  context 'Testing the factorial method' do
```
Comment (SajjadAhmad14) [inline spec/solver_test.rb]: Please add a negative case that raises an exception as an expectation.

File: README.md [hunk 2]
```diff
@@ -1 +1,71 @@
...
+- ruby >= 3.0.1
+
+### Install
+
+- Clone the project using 'git clone https://github.com/tiagorahal/TDD-project.git'
+
+### Run tests
+
+- Use the terminal to open the project
+- run rspecc to check tests
```
Comment (SajjadAhmad14) [inline README.md]: Rspec command does not run the tests. Kindly see the screenshot below:
test

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Tiago and Erick, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
GitHub flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**